docs/spec/overload.rst

```diff
@@ -99,8 +109,349 @@ variable is not sufficient.
 
 Another important difference between type variables such as ``AnyStr``
 and using ``@overload`` is that the prior can also be used to define
-constraints for generic class type parameters.  For example, the type
+constraints for generic class type parameters. For example, the type
 parameter of the generic class ``typing.IO`` is constrained (only
 ``IO[str]``, ``IO[bytes]`` and ``IO[Any]`` are valid)::
 
   class IO(Generic[AnyStr]): ...
+
+
+Invalid overload definitions
```

Comment:
Should having a non-empty body for an ``@overload``-decorated definition be an error? E.g.,
```
@overload
def f(x: int) -> int:
    return x
@overload
def f(x: str) -> str:
    return x
def f(x):
    return x
```
The first two implementations are pointless and potentially misleading.

Reply:
Traditionally, it hasn't been flagged as an error. It will not lead to unsoundness or a runtime exception, so my inclination is to not mandate that type checkers flag it. Arguably, this is more appropriate for a linter. I guess I don't feel that strongly about it one way or another.

I'll note that the typing spec currently doesn't clearly define what "empty body" means, and mypy and pyright differ slightly on their definitions of this. This is on my list of items to discuss and unify, but it's been lower priority.

Reply:
Fair enough! I don't feel strongly about this either, just a passing thought.